app/lib/actions/katello/organization/create.rb

```diff
@@ -3,30 +3,33 @@ module Katello
     module Organization
       class Create < Actions::EntryAction
         def plan(organization)
-          organization.setup_label_from_name
-          organization.create_library
-          organization.create_anonymous_provider unless organization.anonymous_provider
-          organization.create_redhat_provider unless organization.redhat_provider
-          cp_create = nil
+          return if organization.created_in_katello
 
-          organization.save!
+          creator = ::Katello::OrganizationCreator.new(organization)
+          creator.seed!
 
           sequence do
-            cp_create = plan_action(Candlepin::Owner::Create,
-                                    label:  organization.label,
-                                    name: organization.name)
-            plan_action(Katello::Environment::LibraryCreate, organization.library)
-          end
-          if cp_create
-            action_subject organization, label: cp_create.output[:response][:key]
-          else
-            action_subject organization
+            unless organization.candlepin_owner_exists?
+              plan_action(Candlepin::Owner::Create,
+                label: organization.label,
+                name: organization.name
+              )
+            end
+
+            if creator.library_view.cp_environment_id(creator.library_environment).nil?
+              plan_action(Katello::ContentView::AddToEnvironment, creator.library_cvv, creator.library_environment)
+            end
           end
+
           plan_self(:organization_id => organization.id)
         end
 
         def run
-          ::Organization.find_by(:id => input[:organization_id]).try(:load_debug_cert)
+          org = ::Organization.find_by(:id => input[:organization_id])
+          # do we need to create the debug cert here?
+          # why was it written with try...
+          org.try(:load_debug_cert)
```

Comment:
I did a `rake katello:reset` (seeds completed with no errors) and then attempted to load the web UI.

On the first attempt to load `/` I'm getting an Active Record error `Organization with id Default_Organization could not be found.`  I then refresh the browser and the login page shows, and everything works fine.  Here's what the dynflow console looked like:

![org_create_error](https://user-images.githubusercontent.com/22042343/114231368-5bb1b500-9948-11eb-9d72-2c76b5757c29.png)

and here's more of the stack:
```
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/abstract_response.rb:249:in
  `exception_with_response'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/abstract_response.rb:129:in
  `return!'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/request.rb:836:in
  `process_result'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/request.rb:743:in
  `block in transmit'"
- "/opt/rh/rh-ruby25/root/usr/share/ruby/net/http.rb:910:in `start'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/request.rb:727:in
  `transmit'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/request.rb:163:in
  `execute'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/request.rb:63:in
  `execute'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/rest-client-2.1.0/lib/restclient/resource.rb:69:in
  `post'"
- "/home/vagrant/katello/app/lib/katello/http_resource.rb:101:in `post'"
- "/home/vagrant/katello/app/lib/katello/resources/candlepin/owner.rb:94:in `generate_ueber_cert'"
- "/home/vagrant/katello/app/models/katello/glue/candlepin/owner.rb:56:in `generate_debug_cert'"
- "/home/vagrant/katello/app/models/katello/glue/candlepin/owner.rb:62:in `rescue
  in load_debug_cert'"
- "/home/vagrant/katello/app/models/katello/glue/candlepin/owner.rb:59:in `load_debug_cert'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/activesupport-6.0.3.6/lib/active_support/core_ext/object/try.rb:15:in
  `public_send'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/activesupport-6.0.3.6/lib/active_support/core_ext/object/try.rb:15:in
  `try'"
- "/home/vagrant/katello/app/lib/actions/katello/organization/create.rb:31:in `run'"
- "/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/dynflow-1.4.7/lib/dynflow/action.rb:571:in
  `block (3 levels) in execute_run'"
```

From the Dynflow execution plan it looks like `Actions::Katello::Organization::Create` is planned as a concurrence with the two actions in `plan`, meaning it's trying to run `run` before the Candlepin actions are completed.

Reply:
Good observation. Should be fixed

Reply:
indeed it appears so! Resolving